core/block.go

```diff
@@ -138,15 +139,39 @@ func VerifyBlockHash(b *Block, network utils.Network) error {
 func blockHash(b *Block, network utils.Network, overrideSeqAddr *felt.Felt) (*felt.Felt, error) {
 	metaInfo := networkBlockHashMetaInfo(network)
 
+	blockVersion, err := blockVersion(b)
+	if err != nil {
+		return nil, err
+	}
+
 	if b.Number < metaInfo.First07Block {
-		return pre07Hash(b, network.ChainID())
+		return pre07Hash(b, network.ChainID(), blockVersion)
 	}
-	return post07Hash(b, overrideSeqAddr)
+	return post07Hash(b, overrideSeqAddr, blockVersion)
+}
+
+// blockVersion gets the block/starknet version, defaulting to "0.0.0" if it's not specified.
+func blockVersion(b *Block) (*semver.Version, error) {
+
+	var starknetVersion string
+
+	// Account for the fact that some feeders have not specified the block/starknet version
+	if b.Header.ProtocolVersion == "" {
+		starknetVersion = "0.0.0"
+	} else {
+		starknetVersion = b.Header.ProtocolVersion
+	}
+	blockVersion, err := semver.NewVersion(starknetVersion)
+	if err != nil {
+		return nil, err
+	}
+	return blockVersion, nil
+
```

Comment:
we already have a function that parses and handles some quirks that starknet block versions have. 
Please extract the functionality you need from that function in to a separate function and use that in both places.

https://github.com/NethermindEth/juno/blob/c3e110e2823b1ee80d1136a53b657b6c273f7068/blockchain/blockchain.go#L44-L65

Reply:
I've extracted the important parts of checkBlockVersion() into a new function that determines if the blocks protocol version is above 0.11.0 blockVersionPost011()